source/agora/common/Ensure.d

```diff
@@ -47,14 +47,49 @@ private final class FormattedException : Exception
     {
         return this.buffer[0 .. this.end];
     }
+
+    /// Overrides `Throwable.toString` because it uses `msg`...
+    public override void toString (scope void delegate(in char[]) sink) const scope
+    {
+        import core.internal.string : unsignedToTempString;
+
+        char[20] tmpBuff = void;
```

Comment:
```suggestion
        char[20] line_number = void;
```

Reply:
It's almost a pure copy of https://github.com/dlang/druntime/blob/934b4d24bc9a24d93b1d7c380d6155cc0252f9c3/src/object.d#L2439-L2474

Reply:
I think that is why it lacks documentation. Most of Agora code has more descriptive names and comments to make things easier to quickly understand.